packages/core/js-peer/src/index.ts

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ClientConfig, IFluenceClient, RelayOptions, ConnectionState, CallAquaFunctionType, RegisterServiceType } from '@fluencelabs/interfaces';
+import { ClientPeer, makeClientPeerConfig } from './clientPeer/ClientPeer.js';
+import { callAquaFunction } from './compilerSupport/callFunction.js';
+import { registerService } from './compilerSupport/registerService.js';
+import { MarineBackgroundRunner } from './marine/worker/index.js';
+// @ts-ignore
+import { BlobWorker } from 'threads';
+import * as fs from 'fs';
+import process from 'process';
+import { Buffer } from 'buffer';
+import { doRegisterNodeUtils } from './services/NodeUtils.js';
+import { createRequire } from 'module';
+
+const WORKER_VERSION = '__WORKER_VERSION__';
+const MARINE_VERSION = '__MARINE_VERSION__';
+const AVM_VERSION = '__AVM_VERSION__';
+// Override if necessary
+const CDN_ROOT = 'https://unpkg.com/';
+
+const isNode = typeof process !== 'undefined' && process?.release?.name === 'node';
+
+async function fetchResource(packageName: string, assetPath: string, version: string) {
+    if (isNode) {
```

Comment:
if you do `!isNode` and an early return with fetch here - you can avoid extra nesting for the rest of the code in this function

Reply:
Particularly in this place i feel like explicit 'else' clause visibly splits the code to 2 categories - for node and for browser. If we don't have specific standart about writing code, we can use any construction.